### ISSUE TITLE
pretrained model--RRDB_PSNR_x4   is necessary when i train ? 

### ISSUE THREAD

**Issue by ZZprogrammer** [xinntao/BasicSR#81]
hi ,  i just trained a model without pretrained model, and i got a bad result. i want to know if i must use the RRDB_PSNR_x4 model?

and i when prepared the training dataset, i use the extract_subimgs_single.py to devide image to sub.  but there are too many images. so i choose nearly 9700 images for training.   maybe this is one of the reason that i got a bad result?

**Comment by xinntao** [comment]
Do you train with gan loss?
If yes, pre-training could accelerate your training, otherwise, it needs more training. 

Too few images may also result in your problem. Why not with the full datasets? It will not increase your training time with the full dataset.

**Comment by ZZprogrammer** [comment]
Thank you for answering me. And i have a another question. Is the pretrained model just for accelerating training? i trained a new model with pretrained model, but i found that the PSNR still maintained at 24 to 25 when validating on set14. 

i think i have already trained the model with gan loss. And here are my settings when i trained. is that ok?

 "name": "002_RRDB_ESRGAN_x4_DIV2K" //  please remove "debug_" during training
  , "use_tb_logger": true
  , "model":"srragan"
  , "scale": 4
  , "gpu_ids": [0]   



  , "train": {
    "lr_G": 1e-4
    , "weight_decay_G": 0
    , "beta1_G": 0.9
    , "lr_D": 1e-4
    , "weight_decay_D": 0
    , "beta1_D": 0.9
    , "lr_scheme": "MultiStepLR"
    , "lr_steps": [50000, 100000, 200000, 300000]
    , "lr_gamma": 0.5

    , "pixel_criterion": "l1"
    , "pixel_weight": 1e-2
    , "feature_criterion": "l1"
    , "feature_weight": 1
    , "gan_type": "vanilla"
    , "gan_weight": 5e-3

**Comment by xinntao** [comment]
Pre-trained model can provide a good start point.

If you train with GAN loss, the PSNR cannot reflect the visual quality.

**Comment by ZZprogrammer** [comment]
i test your pretrained model , but still got a bad result. i dont know why.  now i m training with the whole DIV2K trainsets. i expect a good result.

**Comment by xinntao** [comment]
What dataset did you test on and which pre-trained model you used?

**Comment by ZZprogrammer** [comment]
RRDB _ESRGAN_x4    and i use the trainset BSDS100 and Set5, the average PSNR are 25.32 and 30.47. in your paper , they should be 27.85 and 32.73 .  and i dont know whether i test correctly.

**Comment by xinntao** [comment]
OK, I see. 
You misunderstood something. PSNR-oriented and perceptual-oriented methods are different.  There are two models. RRDB_PSNR and RRDB_ESRGAN. the first is for PSNR metric, which produced the 32.73 dB results on Set5 datasets. The second is for visual quality, whose PSNR is much lower.

**Comment by ZZprogrammer** [comment]
...i m new to this. sorry about that. it's my mistake.  but i still have a question. how we can evaluate the visual quality. is there a value can judge?

**Comment by xinntao** [comment]
Actually, there is no stand or widely accepted metric. Perpetual index is a metric used in PIRM18 competition. 

Info can be found:  https://github.com/xinntao/ESRGAN/blob/master/QA.md

**Comment by ZZprogrammer** [comment]
thanks a lot !!!

**Comment by lingwei0816** [comment]
Hi，xinntao,Thanks for your great work.
When I trained your code ESRGAN, it‘s loss  not converge. I have a question,did the loss function converge in the model that you finally trained?such as GAN loss,Do you have good advice when training?

**Comment by xinntao** [comment]
For SRGAN/ESRGAN training, the GAN loss fluctuates and does not guarantee to decrease. 
The best practice is to visualize the validation images.

**Comment by lingwei0816** [comment]
OK,thanks a lot!

**Comment by 924175302** [comment]
@xinntao
Hello, I would like to ask how can you visualize PSNR in your code, I have been looking for a long time? Then I still have a few questions to ask you. After training, how to generate a high-resolution image of a low-resolution image without a high-resolution image. It seems that high-resolution and corresponding information is required in test-yml Low resolution maps can run?

**Comment by cuhkszmike** [comment]
@924175302 
The testing part requires GT images just for calculating the PSNR and SSIM, if you just want to test on low resolution images without providing their corresponding high resolution images, you can just modify the functions in util/utils, which used to calculate PSNR and SSIM, to stop their calculation. Then you can run the test without bugs. However, this modification still need to take some time to read the high resolution images, though you can provide any other images instead. 

Actually, there is a mode 'LQ' in test-yml files, though I failed to use it too.
